doc/adrs/138_add_set_collapsing_state_operation_in_view_dsl.adoc

```diff
@@ -0,0 +1,25 @@
+= ADR-138 Add Set Collapsing State Operation in View DSL
+
+== Context
+
+Sirius Web doesn't provide a way for specifiers  to let specifiers define tools that collapse/expand nodes as part of other operations (e.g. to expand a collapsed node when a child node is created). We will add a new _Set Collapsing State_ operation that can be used in tools to expand/collapse nodes as part of the tool execution process.
+
+== Decision
+
+We will add a new `SetCollapsingState` class in the diagram metamodel. This class extends `Operation`, and contains 2 fields:
+
+- `graphicalCandidateExpression : InterpretedExpression`: returns the list of graphical elements to expand/collapse
+- `collapsingState : InterpretedExpression` : returns the CollapsingState to set for the candidates
+
+We will create a new `ViewSetCollapsingStateRequest` class in `org.eclipse.sirius.components.diagrams` with an `elementId` and `collapsingState` attributes. We will update the `DiagramContext` to store these requests.
```

Comment:
Do not make any reference to operations, the view DSL or anything like that in `sirius-components-diagrams`. Diagrams are supposed to be usable programmatically without expressions, without odesigns, and without the view DSL.

Reply:
Why create an entirely new mechanism to update the collapsing state instead of building on the existing one? Why this `ViewSetCollapsingStateRequest` instead of reusing `UpdateCollapsingStateEvent` which already as the same exact properties?

Reply:
I am not sure I fully understand how to address the second comment. I can indeed reuse `UpdateCollapsingStateEvent`, but should it be stored in a collection like the `ViewCreationRequests`?
What drove me away from `UpdateCollapsingStateEvent` is that it is used in the `UpdateCollapsingStateHandler` to set `DiagramContext#diagramEvent`. I can't do this in this context because I may have multiple `UpdateCollapsingStateEvents`, and in any case the `DiagramContext#diagramEvent` will be overriden by the `InvokeSingleClickHandler` at the end of the tool execution.

I think I am missing some elements on how `DiagramContext#diagramEvent` is used, and when should a tool set it instead of using a mechanism like the one used for `ViewCreationRequests`